报错内容
- paddle
```
W0728 01:58:02.178110  1445 gpu_resources.cc:114] Please NOTE: device: 0, GPU Compute Capability: 8.0, Driver API Version: 12.0, Runtime API Version: 12.6
Traceback (most recent call last):
  File "/data/workspace/cusolver/lstsq.py", line 23, in <module>
    result = paddle.linalg.lstsq(a, b, rcond=1e-15, driver="gels")
  File "/data/Paddle/build/python/paddle/tensor/linalg.py", line 4828, in lstsq
    solution, residuals, rank, singular_values = _C_ops.lstsq(
OSError: (External) CUSOLVER error(3). 
  [Hint: Please search for the error code(3) on website (https://docs.nvidia.com/cuda/cusolver/index.html#cuSolverSPstatus) to get Nvidia's official solution and advice about CUSOLVER Error.] (at /data/Paddle/paddle/phi/kernels/impl/lstsq_kernel_impl.h:204)
```

- torch
```
test begin: paddle.linalg.lstsq(Tensor([253522376, 9],"float32"), Tensor([253522376, 5],"float32"), rcond=1e-15, driver="gels", )
[torch error] paddle.linalg.lstsq(Tensor([253522376, 9],"float32"), Tensor([253522376, 5],"float32"), rcond=1e-15, driver="gels", ) 
 cusolver error: CUSOLVER_STATUS_INVALID_VALUE, when calling `cusolverDnSormqr_bufferSize(handle, side, trans, m, n, k, A, lda, tau, C, ldc, lwork)`. This error may appear if the input matrix contains NaN. If you keep seeing this error, you may use `torch.backends.cuda.preferred_linalg_library()` to try linear algebra operators with other supported backends. See https://pytorch.org/docs/stable/backends.html#torch.backends.cuda.preferred_linalg_library
Traceback (most recent call last):
  File "/data/PaddleAPITest/tester/accuracy.py", line 93, in test
    exec(code.core_compiled, exec_globals, exec_locals)
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_device.py", line 104, in __torch_function__
    return func(*args, **kwargs)
torch._C._LinAlgError: cusolver error: CUSOLVER_STATUS_INVALID_VALUE, when calling `cusolverDnSormqr_bufferSize(handle, side, trans, m, n, k, A, lda, tau, C, ldc, lwork)`. This error may appear if the input matrix contains NaN. If you keep seeing this error, you may use `torch.backends.cuda.preferred_linalg_library()` to try linear algebra operators with other supported backends. See https://pytorch.org/docs/stable/backends.html#torch.backends.cuda.preferred_linalg_library
W0728 08:42:25.010032 68628 gpu_resources.cc:114] Please NOTE: device: 0, GPU Compute Capability: 8.0, Driver API Version: 12.0, Runtime API Version: 12.6

```

--- a/paddle/phi/kernels/gpu/lstsq_kernel.cu
+++ b/paddle/phi/kernels/gpu/lstsq_kernel.cu
@@ -16,6 +16,7 @@
 #include <algorithm>
 #include <complex>
 
+#include "paddle/phi/backends/gpu/cuda/cudnn_workspace_helper.h"
 #include "paddle/phi/backends/gpu/gpu_context.h"
 #include "paddle/phi/core/kernel_registry.h"
 #include "paddle/phi/kernels/full_kernel.h"
@@ -62,23 +63,8 @@ void LstsqKernel(const Context& dev_ctx,
     return;
   }
 
-  int64_t largest_matrix = (1LL << 31) - 1;
-  int64_t matrix_size_x = 1;
-  int64_t matrix_size_y = 1;
-  for (int i = 0; i < x.dims().size(); ++i) {
-    matrix_size_x *= x.dims()[i];
-  }
-  for (int i = 0; i < y.dims().size(); ++i) {
-    matrix_size_y *= y.dims()[i];
-  }
-  int64_t matrix_size = std::max(matrix_size_x, matrix_size_y);
-  PADDLE_ENFORCE_LE(
-      matrix_size,
-      largest_matrix,
-      ::common::errors::PreconditionNotMet(
-          "Matrix size too large for LeaST SQuares solution. Maximum "
-          "allowed size is 2 ^ 31 - 1 elements, but got %lld",
-          matrix_size));
+  CUDNN_ENFORCE_TENSOR_SIZE_SUPPORTED(x);
+  CUDNN_ENFORCE_TENSOR_SIZE_SUPPORTED(y);
 
   auto x_dims = x.dims();
   auto y_dims = y.dims();

